mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -1084,8 +1084,8 @@ static LogicalResult alignedConversionPrecondition(PatternRewriter &rewriter,
   unsigned srcElemBitwidth = srcType.getElementTypeBitWidth();
   unsigned dstElemBitwidth = dstType.getElementTypeBitWidth();
 
-  // Only {s}i4 -> (size_of({{s}i/f}) >= 8) are supported for now.
-  if (srcElemBitwidth != 4 || dstElemBitwidth < 8 ||
+  // Only {s}i4/i2 -> (size_of({{s}i/f}) >= 8) are supported for now.
+  if ((srcElemBitwidth != 4 && srcElemBitwidth != 2) || dstElemBitwidth < 8 ||
       (dstElemBitwidth % srcElemBitwidth) != 0)
     return rewriter.notifyMatchFailure(op, "Not a supported aligned case");
```

Comment:
This statement is becoming more complicated, and it is not easy to parse if the comment match the check or not. How about we break it into three if-statement and collapse the comment into the error message? E.g.,

```cpp
// I feel that it is a requirement for emulation, so we can just say that it must be greater than or equal to 8
if (dstElemBitwidth < 8)
  return rewriter.notifyMatchFailure(op, "the bitwidth of dstType must be greater than or equal to 8");
if (dstElemBitwidth % srcElemBitwidth != 0)
  return ... "unaligned cases are not supported"
if (srcElemBitwidth != 2 && srcElemBitwidth != 4)
  return ... "only support src bitwidth to be 2 or 4 at this moment"
}
```